README.md

```diff
@@ -17,23 +17,91 @@ original Secret from the SealedSecret.
 See https://github.com/bitnami-labs/sealed-secrets/releases for the latest
 release.
 
-```sh
+
+### GKE Requirements
+
+If installing on a GKE cluster, a ClusterRoleBinding may be needed to successfully deploy the controller in the final command.  Replace <your-email> with a valid email, and then deploy the cluster role binding:
+ 
+```bash
+USER_EMAIL=<your-email>
+kubectl create clusterrolebinding $USER-cluster-admin-binding --clusterrole=cluster-admin --user=$USER_EMAIL
+```
+ 
+If you are using a **private GKE cluster**, you are required to create a Master-to-Node firewall rule to allow GKE to communicate to the kubeseal container endpoint port tcp/8080.
+ 
+```bash
+CLUSTER_NAME=foo-cluster
+gcloud config set compute/zone your-zone-or-region
+```
+
+Get the `MASTER_IPV4_CIDR`.
+
+```bash
+MASTER_IPV4_CIDR=$(gcloud container clusters describe $CLUSTER_NAME \
+  | grep "masterIpv4CidrBlock: " \
+  | awk '{print $2}')
+```
+
+Get the `NETWORK`.
+
+```bash
+NETWORK=$(gcloud container clusters describe $CLUSTER_NAME \
+  | grep "^network: " \
+  | awk '{print $2}')
+```
+
+Get the `NETWORK_TARGET_TAG`.
+
+```bash
+NETWORK_TARGET_TAG=$(gcloud compute firewall-rules list \
+  --filter network=$NETWORK --format json \
+  | jq ".[] | select(.name | contains(\"$CLUSTER_NAME\"))" \
+  | jq -r '.targetTags[0]' | head -1)
+```
+
+Check the values.
+
+```bash
+echo $MASTER_IPV4_CIDR $NETWORK $NETWORK_TARGET_TAG
+
+# example output
+10.0.0.0/28 foo-network gke-foo-cluster-c1ecba83-node
+```
+
+Create the firewall rule.
+
+```bash
+gcloud compute firewall-rules create gke-to-kubeseal-8080 \
+  --network "$NETWORK" \
+  --allow "tcp:8080" \
+  --source-ranges "$MASTER_IPV4_CIDR" \
+  --target-tags "$NETWORK_TARGET_TAG" \
+  --priority 1000
+```
+
+### Installing the cli
+
+Install the kubeseal cli into `/usr/local/bin/kubeseal`.
+
+```bash
 $ release=$(curl --silent "https://api.github.com/repos/bitnami-labs/sealed-secrets/releases/latest" | sed -n 's/.*"tag_name": *"\([^"]*\)".*/\1/p')
 
-# Install client-side tool into /usr/local/bin/
-$ GOOS=$(go env GOOS)
-$ GOARCH=$(go env GOARCH)
-$ wget https://github.com/bitnami-labs/sealed-secrets/releases/download/$release/kubeseal-$GOOS-$GOARCH
-$ sudo install -m 755 kubeseal-$GOOS-$GOARCH /usr/local/bin/kubeseal
+# See the release binary versions at https://github.com/bitnami-labs/sealed-secrets/releases/
+RELEASE_BIN_URL=https://github.com/bitnami-labs/sealed-secrets/releases/download/v0.8.3/kubeseal-linux-amd64
```

Comment:
I see two problems with this change in the instructions:

1. This has to be updated for every release (and we risk forgetting it)
2. Not everybody runs linux on amd64

To deal with (1) so far I opted for just inlining installation instructions in each release notes (so that they include the correct version).

I assume that you prefer ditching `go env ...` because not everybody has the go toolchain installed.
What if we just point to the latest release page here?

Reply:
> I assume that you prefer ditching go env ... because not everybody has the go toolchain installed.

Yes!

> What if we just point to the latest release page here?

I think that is a better idea to avoid having to maintain the main README. What do you think about the one-liner curl install though?

Reply:
you mean 

```
RELEASE_BIN_URL=https://github.com/bitnami-labs/sealed-secrets/releases/download/v0.8.3/kubeseal-linux-amd64
curl -fsSL -o /usr/local/bin/kubeseal "$RELEASE_BIN_URL" && chmod +x /usr/local/bin/kubeseal
```

(probably should be `sudo curl` to work for all users)

as opposed to:

```
wget https://github.com/bitnami-labs/sealed-secrets/releases/download/v0.8.3/kubeseal-linux-amd64 -O kubeseal
sudo install -m 755 kubeseal /usr/local/bin/kubeseal
```

?

Yeah I have mixed feelings. wget vs curl? sudo curl vs sudo only the file copy? write directly vs tmp file in local dir? ...

I don't know, I'm tempted to just write a bash installer you download and run. That would also cover the arch/os detection and provide a simple

e.g.:

```
wget https://github.com/bitnami-labs/sealed-secrets/releases/download/v0.8.3/kubeseal-install.sh && sh kubeseal-install.sh
```

wdyt?